futures-test/src/assert.rs

```diff
@@ -0,0 +1,126 @@
+use futures_core::stream::Stream;
+use std::marker::Unpin;
+
+#[doc(hidden)]
+pub fn assert_is_unpin_stream<S: Stream + Unpin>(_: &mut S) {}
+
+/// Assert that the next poll to the provided stream will return
+/// [`Poll::Pending`][futures_core::task::Poll::Pending].
+///
+/// # Examples
+///
+/// ```
+/// #![feature(async_await, futures_api, pin)]
+///
+/// use futures::stream;
```

Comment:
Our code examples usually have no free line between `#![feature]` and `use` statements